genson/src/main/java/com/owlike/genson/stream/JsonWriter.java

```diff
@@ -241,10 +241,20 @@ public JsonWriter writeValue(int value) {
   }
 
   public JsonWriter writeValue(final double value) {
-    checkValidJsonDouble(value);
-    clearMetadata();
-    beforeValue();
-    writeToBuffer(Double.toString(value), 0);
+      boolean allowExtendedNumberTypes=true;  //this should come from gensonbuilder?
+	  clearMetadata();
+      beforeValue();
+      
+	  if (allowExtendedNumberTypes==true)
+	  {
+		 String writethis = getValidJsonDouble(value);
+		 writeToBuffer(writethis,0);
+	  }
+	  else
+	  {	  
+		checkValidJsonDouble(value);	
+		writeToBuffer(Double.toString(value), 0);
+	  }
```

Comment:
Use proper indentation (two spaces instead of four), clearMetada and the if else are not correctly indented relative to the rest. Put the { on the same line as the expression. Put the else on the same line as the closing } from the if.